From 6b226b376061b05c2a2804525860e747c7b45069 Mon Sep 17 00:00:00 2001
From: sebastien baudouin <sebastien.baudouin@windriver.com>
Date: Tue, 28 Apr 2015 16:57:44 +0200
Subject: [PATCH 6/6] Launcher2: Simplify the OpenGLes part

This commit is there to reduce MIPS needed for OpenGL-ES . Those modifications
are needed in case of Qemu machine which is emulating GPU and so consuming a lot
of mips.
---
 content/ListItem.qml   | 11 ++++++++++-
 content/NMapEffect.qml | 26 +++++++++++++++++++++-----
 gdp-hmi-launcher2.qml  |  8 ++++----
 3 files changed, 35 insertions(+), 10 deletions(-)

diff --git a/content/ListItem.qml b/content/ListItem.qml
index 2ff531e..3b7f04e 100755
--- a/content/ListItem.qml
+++ b/content/ListItem.qml
@@ -27,7 +27,7 @@ Item {
     z: PathView.iconZ
     scale: PathView.iconScale
 
-    Image {
+    /*Image {
         id: shadowSource
         source: model.shadowIcon
         width: nmapItem.width
@@ -50,6 +50,15 @@ Item {
         elementPositionY: root.y
         colorizeAmount: fogAmount
         diffuseBoost: 0.5
+    }*/
+
+    Image {
+        id: nmapItem
+        source: model.sourceIcon
+        width: root.width
+        height: root.height
+        opacity: 1
+        scale: 1
     }
 
     MouseArea {
diff --git a/content/NMapEffect.qml b/content/NMapEffect.qml
index e7fbf06..980e086 100755
--- a/content/NMapEffect.qml
+++ b/content/NMapEffect.qml
@@ -26,7 +26,7 @@ ShaderEffect {
     // Boost diffuse effect of this item
     property real diffuseBoost: 0.0
     // Light intensity from source or alternatively custom intensity for this item
-    property real lightIntensity: lightSource.lightIntensity
+    //property real lightIntensity: lightSource.lightIntensity
     // Switch x-coordinate of normal mapped image
     property bool switchX: false
     // Switch y-coordinate of normal mapped image
@@ -42,8 +42,8 @@ ShaderEffect {
     property real originalHeight: sourceImageItem.sourceSize.height
 
     /* Private */
-    property real _lightPosX: lightSource.lightPosX / lightSource.width * (lightSource.width/root.width) - elementPositionX/root.width
-    property real _lightPosY: lightSource.lightPosY / lightSource.height * (lightSource.height/root.height) - elementPositionY/root.height
+    //property real _lightPosX: lightSource.lightPosX / lightSource.width * (lightSource.width/root.width) - elementPositionX/root.width
+    //property real _lightPosY: lightSource.lightPosY / lightSource.height * (lightSource.height/root.height) - elementPositionY/root.height
     property variant _source: ShaderEffectSource { sourceItem: sourceImageItem; hideSource: true }
     property variant _source2: ShaderEffectSource { sourceItem: normalsourceImageItem; hideSource: true }
 
@@ -61,7 +61,7 @@ ShaderEffect {
         visible: false
     }
 
-    fragmentShader: "
+    /*fragmentShader: "
     varying highp vec2 qt_TexCoord0;
     uniform highp float qt_Opacity;
     uniform sampler2D _source;
@@ -93,5 +93,21 @@ ShaderEffect {
 
         gl_FragColor = color * qt_Opacity;
     }
-    "
+    "*/
+    fragmentShader: "
+      varying highp vec2 qt_TexCoord0;
+      uniform highp float qt_Opacity;
+      uniform sampler2D _source;
+      uniform sampler2D _source2;
+      uniform highp float diffuseBoost;
+      uniform highp float colorizeAmount;
+      uniform highp vec4 colorizeColor;
+      uniform bool switchX;
+      uniform bool switchY;
+      void main(void)
+      {
+        highp vec2 pixPos = qt_TexCoord0;
+        highp vec4 pix = texture2D(_source, pixPos.st);
+        gl_FragColor = pix;
+      }"
 }
diff --git a/gdp-hmi-launcher2.qml b/gdp-hmi-launcher2.qml
index aeb446b..4e6cdf3 100755
--- a/gdp-hmi-launcher2.qml
+++ b/gdp-hmi-launcher2.qml
@@ -50,11 +50,11 @@ Item {
     QtObject {
         id: settings
         property real spotAnimationPosition: 0.0
-        SequentialAnimation on spotAnimationPosition {
+        /*SequentialAnimation on spotAnimationPosition {
             loops: Animation.Infinite
             NumberAnimation { to: 30; duration: 2000; easing.type: Easing.InOutQuad }
             NumberAnimation { to: 0; duration: 2000; easing.type: Easing.InOutQuad }
-        }
+        }*/
     }
 
     Image {
@@ -161,7 +161,7 @@ Item {
         }
     }
 
-    Image {
+    /*Image {
         anchors.bottom: parent.bottom
         anchors.horizontalCenter: parent.horizontalCenter
         anchors.horizontalCenterOffset: 150
@@ -184,7 +184,7 @@ Item {
         height: 200
         rotation: -settings.spotAnimationPosition * 0.8
         opacity: 0.2
-    }
+    }*/
 
     PowerOffItem {
         id: powerOffItem
-- 
1.9.1

